rubix-core/src/main/java/com/qubole/rubix/core/RemoteReadRequestChain.java

```diff
@@ -192,17 +192,28 @@ public ReadRequestChainStats getStats()
   @Override
   public void updateCacheStatus(String remotePath, long fileSize, long lastModified, int blockSize, Configuration conf)
   {
+    RetryingBookkeeperClient client = null;
     try {
-      RetryingBookkeeperClient client = bookKeeperFactory.createBookKeeperClient(conf);
+      client = bookKeeperFactory.createBookKeeperClient(conf);
```

Comment:
You can use try-with-resources to make sure the client is automatically closed once the `try` block has finished.
ex: `try (RetryingBookKeeper client = bookKeeperFactory.createBookKeeperClient(conf)) {`